runtime/test/strategies/convert-constraints-to-connections-tests.js

```diff
@@ -333,13 +333,66 @@ describe('ConvertConstraintsToConnections', async () => {
     assert.equal(results.length, 1);
     assert.deepEqual(results[0].result.toString(), `recipe
   ? #hashtag as handle0 // S {}
-  create as handle1 // S {}
+  create #trashbag as handle1 // S {}
   A as particle0
     o -> handle0
   B as particle1
     i -> handle1`);
   });
 
+  it('connects existing particles to tags', async () => {
+    let recipes = (await Manifest.parse(`
+    particle A
+      out S {} o
+    particle B
+      out S {} i
+    recipe
+      ? #hashtag
+      A.o -> #hashtag
+      #trashbag <- B.i
+      A
+      B
+    `)).recipes;
+    let inputParams = {generated: [{result: recipes[0], score: 1}]};
+    let cctc = new ConvertConstraintsToConnections({pec: {}});
+    let results = await cctc.generate(inputParams);
+    assert.equal(results.length, 1);
+    assert.deepEqual(results[0].result.toString(), `recipe
+  ? #hashtag as handle0 // S {}
+  create #trashbag as handle1 // S {}
+  A as particle0
+    o -> handle0
+  B as particle1
+    i -> handle1`);
+  });
+
+  it(`doesn't attempt to duplicate existing connections to tags`, async () => {
+    let recipes = (await Manifest.parse(`
+    particle A
+      out S {} o
+    particle B
+      out S {} i
+    recipe
+      ? #hashtag as handle0
+      A.o -> #hashtag
+      #trashbag <- B.i
```

Comment:
#trashbag tag here is completely lost, i see that this is intentional, but i don't understand why. maybe another strategy later could have made something useful out of it?